_posts/2020-11-xx-azure-devops-pipeline-as-code.md

```diff
@@ -0,0 +1,234 @@
+
+Source: [Converting Classic Azure DevOps Pipelines to YAML](https://devblogs.microsoft.com/premier-developer/converting-classic-azure-devops-pipelines-to-yaml/)
+
+## Classic vs YAML Pipeline
+
+---
+**Rule of thumb**
+
+For new pipelines, use YAML. For existing pipelines, keep classic (but consider converting to YAML).
+
+---
+
+The biggest pro for YAML pipelines is that they will be stored into version control.
+And thus important rule "keep everything in version control" can be achieved.
+That is a big lack of classic pipelines.
+
+YAML pipelines support also better different versions of the pipeline.
+If we want to restore the older version of the code, the pipeline's code will also be restored.
+But with classic pipelines, this doesn't happen: we have to manually search for the version we need to restore.
+This is a big thing if the pipeline has changed significantly between versions.
+
+Another good side with the version control is when making some changes to the pipeline with YAML, we can first make changes in a feature branch and test the pipeline there.
+And when we have tested the pipeline, we can merge it.
+This isn't impossible with classic pipelines.
+But it is a more tricky task with classic pipelines as we have to put conditional steps that will run only with a given branch.
+
+There is one pro for the classic pipeline.
+As it is graphical, it is easier to understand especially if we don't use Azure Pipelines so often.
+For the same reason, there is a lower learning curve with classic pipelines than with YAML pipelines.
+
+Feature                         | YAML pipeline                    | Classic pipeline
+--------------------------------|----------------------------------|-----------------
+Can be saved to version control | yes                              | no
+Restore old version             | yes, restored together with code | yes, but restored separately
+Changes in isolation            | yes, with branches               | no
+Graphical view                  | no (yes for settings)            | yes
+Easy to learn                   | no                               | yes
+
+## Conclusion
+
+At first, YAML pipelines look difficult compared to classic pipelines.
+Some of us might doubt if it is too difficult to start using YAML pipelines.
+Fortunately, there isn't a need to know everything by heart.
+There is some graphical help in the Azure DevOps web portal and IntelliSense will also help with the web portal and Visual Studio Code.
+If you are in doubt about YAML pipelines, I recommend to give them a chance.
+They can be the next upgrade for your pipelines and efficiency.
+
+## Further Reading
+
+* [Use Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops)
+* [YAML schema reference](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema)
```

Comment:
Good links. Possibly include a link to another Solita blog that deals with Azure devops? It's not the same viewpoint, but nice extra story on the theme. Cross-linking each others blogs makes them always much more visible. I found this with google search: "solita infrastructure as code" :)

https://data.solita.fi/automatized-code-deployment-from-azure-devops-to-databricks/

Reply:
Thanks! I added that link. I searched only from dev blog and didn't realize we have other blogs also. Good reminder for me.